### PR TITLE
WS list: gracefully handle workspaces with incomplete definitions [GAWB-2872]

### PR DESCRIPTION
**Pull request by aednichols**
@zarsky-broad and I discovered that the workspace response, or some unknown transformation thereof taking place on the frontend, may sometimes contain illegal workspace definitions, along the lines of
```
{
  "workspaceSubmissionStats": {
    "lastSuccessDate": "2017-05-15T15:03:11.269Z",
    "lastFailureDate": "2017-05-15T15:03:01.503Z",
    "runningSubmissionsCount": 0
  },
  "accessLevel": "PROJECT_OWNER",
  "owners": ["test.firec@gmail.com"],
  "public": false,
  "workspace": {}
}
```
Instead of throwing `a.toLowerCase`, we now display an inert dummy. 

This feels like the least surprising thing we could do; we thought about filtering them out, but that could be even more confusing (_"I have 5 workspaces, why do I only see 4?"_ vs. _"Huh, something is seriously wrong with that one workspace."_).

Looks like this:

screenshot-2017-12-14 firecloud broad institute

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspaces_list.cljs [hunk 1]
```diff
@@ -412,7 +412,15 @@
    (fn [{:keys [props state]}]
      (let [{:keys [server-response]} @state
            workspaces (map
-                       (fn [ws] (assoc ws :status (common/compute-status ws)))
+                       (fn [ws]
+                         ;; GAWB-2872
+                         (if (or (nil? (:name (:workspace ws))) (nil? (:namespace (:workspace ws))))
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/page/workspaces_list.cljs]: you should try: 
```cljs
(let [{:keys [name namespace]} (:workspace ws)]
  (if (and name namespace)
    (assoc ws ...
```

**Review by zarsky-broad — APPROVED** [review]

**Comment by zarsky-broad** [comment]
We should fix unit tests before we merge this.

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
Having trouble telling whether they're my code or just noise

**Comment by davidangb** [comment]
Some questions:

1. How could I reproduce this locally, so I can see it in action?
2. In lieu of the above ... what happens if I click on one of those workspaces? They say No Access, so I assume nothing ... but wanna check. 
3. In lieu of the above ... any impact to search/filters, pagination, etc?

**Comment by breilly2** [comment]
Let's chat about this. This may eliminate the errors, but it doesn't address the underlying problem and, I fear, takes some of the heat off of actually fixing it. We need to add something here to gather information so that we can better understand what's happening.

**Comment by aednichols** [comment]
@davidangb and @breilly2:

We do not have reliable repro steps nor a root cause. To simulate the bug and examine the behavior of this PR's code, you can replace
```
(get-in server-response [:workspaces-response :parsed-response])
```
with
```
(concat (get-in server-response [:workspaces-response :parsed-response]) (vec (repeat 10 {})))
```
in `workspace_list.cljs`. (You get 10 dummy workspaces at the end of your WS list; can also make the same change on top of `develop` and compare behaviors.)

I manually tested paging, search, and filtering - no unexpected behaviors. Filtering specifically works pretty sensibly, the "dead" workspaces fall under, in respective categories, "No Access", "Exception", and "Un-published".

**Review by davidangb — APPROVED** [review]
:+1: I'm okay with this; I think Brian has a point about finding the culprit but think we could push this in. If the problem goes away, we have a pretty good idea of what's happening.

**Pull request merged**